File: D_Replacing/v1.cpp
#include<iostream>
#include<map>
using namespace std;

int main() {
	long len;
    cin >> len;
	
    long sum = 0;
    map<long, long> numCount;
    for (int i = 0; i < len; i++) {
    	long a;
        cin >> a;
      	sum += a;
        numCount[a] += 1;
    }
  
    long q;
    cin >> q;
    for (long i = 0; i < q; i++) {
    	long b, c;
        cin >> b >> c;
      	
      	long amount = numCount[b];
        sum += amount * (c - b);
        cout << sum << endl;
      	
      	numCount[c] += amount;
      	numCount[b] = 0;
    }
}

File: D_Replacing/v1.js
const fs = require('fs')
const inputs = fs
  .readFileSync('/dev/stdin', 'utf8')
  .trim()
  .split(/\s+/)
  .map(s => parseInt(s))

const len = inputs.shift()
let sum = 0
const numCount = {}
for (let i = 0; i < len; i++) {
  const num = inputs.shift()
  numCount[`${num}`] = (numCount[`${num}`] || 0) + 1
  sum += num
}

const q = inputs.shift()

for (let i = 0; i < q; i += 1) {
  const b = inputs.shift()
  const c = inputs.shift()
  
  const amount = numCount[`${b}`] || 0
  sum += amount * (c - b)
  console.log(sum)
  
  numCount[`${c}`] = (numCount[`${c}`] || 0) + amount
  numCount[`${b}`] = 0
}

